conninfo/src/conninfo-v1.0-wd01.md

```diff
@@ -136,41 +135,50 @@ servers, or containers supporting a peer-to-peer scenario.
+to omit a key from the map. An implementation MAY also include keys of its own.
+
+The receiving partner SHOULD NOT ascribe any semantics to a key not defined by this specification or its
+value, and the sender partner MUST NOT expect a receiver to ascribe any semantics to such a key.
+
+# 3.2.1 connection-info
 
 **Table 1-1. connection-info**
 
 | Name | Description |
 | :--- | :--- |
 | `process-identifier` | The value of the process-identifier MUST be an AMQP string. The purpose of the value is to identify the process running this AMQP host. It is RECOMMENDED that this is an operating system PID. |
 | `network-host` | The value of network-host MUST be an AMQP string. The purpose of the value is to identify the host running the AMQP container in a way that assists those fault finding within a system. This value MAY be a hostname, fully qualified hostname or IP address. The value MAY be network resolvable by the AMQP peer but the peer MUST NOT rely on this. |
-| `amqp-product` | The value of amqp-product MUST be an AMQP string and is a `product-string`. It provides details of the software product(s) that provide the AMQP implementation. It MAY identify multiple products. By convention, the products are listed in order of their significance for identifying the product. |
-| `product` | The value of product MUST be an AMQP string and is a product-string. It provides details of the software product(s) that is the end-user application. It MAY identify multiple products. By convention, the products are listed in order of their significance for identifying the application. |
-| `platform` | The value of platform MUST be an AMQP string and is a product-string. It exists to expose details of a product’s runtime platform. This typically identify the operating system, Java/.NET runtimes etc. It MAY identify multiple products. By convention, the products are listed in order of their significance. |
+| `amqp-component` | The value of `amqp-component` MUST be an AMQP string and is a `product-string`. It provides details of the software components(s) that provide the AMQP implementation. It MAY identify multiple components. By convention, the components are listed in order of their significance. |
+| `application` | The value of `application` MUST be an AMQP string and is a `product-string`. It provides details of the software application(s) that make use of the AMQP components. It MAY identify multiple applications. By convention, the applications are listed in order of their significance. |
+| `platform` | The value of platform MUST be an AMQP string and is a `product-string`. It exists to expose details of the application’s runtime platform. This typically identifies the operating system, Java/.NET runtimes etc. It MAY identify multiple platforms. By convention, the platforms are listed in order of their significance. |
+
+# 3.2.1 product-string
 
-# 3.2 product-string
+The XML notation for this type is:
```

Comment:
I still have some nagging concerns about use of the user-agent.   If a user wished to extract say product and version from the string, he would be forced to write a user-agent parser.  The fact that user-agent allows zero or more comment(s) per product, means this parser would be non-trivial.  The following is a legal user-agent.

```
Mozilla/5.0 (platform; rv:geckoversion) Gecko/geckotrail Firefox/firefoxversion
```

I propose changing `amqp-component` etc to be a *list of strings* where each string would describe a single product.

```
product         = token ["/" product-version]
```

or if we care to keep the comments support (I don't, btw).

```
product         = token ["/" product-version] *[ "(" *( ctext / quoted-pair / comment ) ")"]
```

The advantage is that a trivial parser recover the product name and product version (split on whitespace, split on slash).   It also means if a UI wanted to list the product/version/comment in a table, it can just iterate the list.